### PR TITLE
Fix: StackDriver exporter fails with "One or more Timeseries could not be written"

### LINKED ISSUE #350: stats/exporters: StackDriver exporter fails with "One or more Timeseries could not be written"
Am using OpenCensus-Python 0.1.8

and on trying out the example at https://opencensus.io/guides/integrations/memcached/python/#end-to-end-example

I get this printed to my console, exceptions thrown
```shell
ERROR:root:StackdriverStatsExporter failed to emit data.Dropping 2 objects from queue.
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/opencensus/common/transports/async.py", line 105, in _thread_main
    self.exporter.emit(data)
  File "/usr/local/lib/python2.7/site-packages/opencensus/stats/exporters/stackdriver_exporter.py", line 145, in emit
    self.handle_upload(view_data)
  File "/usr/local/lib/python2.7/site-packages/opencensus/stats/exporters/stackdriver_exporter.py", line 157, in handle_upload
    self.upload_stats(view_data)
  File "/usr/local/lib/python2.7/site-packages/opencensus/stats/exporters/stackdriver_exporter.py", line 166, in upload_stats
    request[CONS_TIME_SERIES])
  File "/usr/local/lib/python2.7/site-packages/google/cloud/monitoring_v3/gapic/metric_service_client.py", line 868, in create_time_series
    request, retry=retry, timeout=timeout, metadata=metadata)
  File "/usr/local/lib/python2.7/site-packages/google/api_core/gapic_v1/method.py", line 139, in __call__
    return wrapped_func(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/google/api_core/retry.py", line 260, in retry_wrapped_func
    on_error=on_error,
  File "/usr/local/lib/python2.7/site-packages/google/api_core/retry.py", line 177, in retry_target
    return target()
  File "/usr/local/lib/python2.7/site-packages/google/api_core/timeout.py", line 206, in func_with_timeout
    return func(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/google/api_core/grpc_helpers.py", line 56, in error_remapped_callable
    six.raise_from(exceptions.from_grpc_error(exc), exc)
  File "/usr/local/lib/python2.7/site-packages/six.py", line 718, in raise_from
    raise value
InvalidArgument: 400 One or more TimeSeries could not be written: Field timeSeries[0].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[0]; Field timeSeries[1].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[1]
ERROR:root:StackdriverStatsExporter failed to emit data.Dropping 10 objects from queue.
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/opencensus/common/transports/async.py", line 105, in _thread_main
    self.exporter.emit(data)
  File "/usr/local/lib/python2.7/site-packages/opencensus/stats/exporters/stackdriver_exporter.py", line 145, in emit
    self.handle_upload(view_data)
  File "/usr/local/lib/python2.7/site-packages/opencensus/stats/exporters/stackdriver_exporter.py", line 157, in handle_upload
    self.upload_stats(view_data)
  File "/usr/local/lib/python2.7/site-packages/opencensus/stats/exporters/stackdriver_exporter.py", line 166, in upload_stats
    request[CONS_TIME_SERIES])
  File "/usr/local/lib/python2.7/site-packages/google/cloud/monitoring_v3/gapic/metric_service_client.py", line 868, in create_time_series
    request, retry=retry, timeout=timeout, metadata=metadata)
  File "/usr/local/lib/python2.7/site-packages/google/api_core/gapic_v1/method.py", line 139, in __call__
    return wrapped_func(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/google/api_core/retry.py", line 260, in retry_wrapped_func
    on_error=on_error,
  File "/usr/local/lib/python2.7/site-packages/google/api_core/retry.py", line 177, in retry_target
    return target()
  File "/usr/local/lib/python2.7/site-packages/google/api_core/timeout.py", line 206, in func_with_timeout
    return func(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/google/api_core/grpc_helpers.py", line 56, in error_remapped_callable
    six.raise_from(exceptions.from_grpc_error(exc), exc)
  File "/usr/local/lib/python2.7/site-packages/six.py", line 718, in raise_from
    raise value
InvalidArgument: 400 One or more TimeSeries could not be written: Field timeSeries[0].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[0]; Field timeSeries[1].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[1]; Field timeSeries[2].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[2]; Field timeSeries[3].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[3]; Field timeSeries[4].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[4]; Field timeSeries[5].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[5]; Field timeSeries[6].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[6]; Field timeSeries[7].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[7]; Field timeSeries[8].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[8]; Field timeSeries[9].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[9]
ERROR:root:StackdriverStatsExporter failed to emit data.Dropping 4 objects from queue.
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/opencensus/common/transports/async.py", line 105, in _thread_main
    self.exporter.emit(data)
  File "/usr/local/lib/python2.7/site-packages/opencensus/stats/exporters/stackdriver_exporter.py", line 145, in emit
    self.handle_upload(view_data)
  File "/usr/local/lib/python2.7/site-packages/opencensus/stats/exporters/stackdriver_exporter.py", line 157, in handle_upload
    self.upload_stats(view_data)
  File "/usr/local/lib/python2.7/site-packages/opencensus/stats/exporters/stackdriver_exporter.py", line 166, in upload_stats
    request[CONS_TIME_SERIES])
  File "/usr/local/lib/python2.7/site-packages/google/cloud/monitoring_v3/gapic/metric_service_client.py", line 868, in create_time_series
    request, retry=retry, timeout=timeout, metadata=metadata)
  File "/usr/local/lib/python2.7/site-packages/google/api_core/gapic_v1/method.py", line 139, in __call__
    return wrapped_func(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/google/api_core/retry.py", line 260, in retry_wrapped_func
    on_error=on_error,
  File "/usr/local/lib/python2.7/site-packages/google/api_core/retry.py", line 177, in retry_target
    return target()
  File "/usr/local/lib/python2.7/site-packages/google/api_core/timeout.py", line 206, in func_with_timeout
    return func(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/google/api_core/grpc_helpers.py", line 56, in error_remapped_callable
    six.raise_from(exceptions.from_grpc_error(exc), exc)
  File "/usr/local/lib/python2.7/site-packages/six.py", line 718, in raise_from
    raise value
InvalidArgument: 400 One or more TimeSeries could not be written: Field timeSeries[0].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[0]; Field timeSeries[1].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[1]; Field timeSeries[2].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[2]; Field timeSeries[3].points had an invalid value: Only one point can be written per TimeSeries per request.: timeSeries[3]
Background thread exited.
Sent all pending spans.
```

### PR DESCRIPTION
**Pull request by mayurkale22**
Fixes #350

### CODE REVIEW

File: examples/stats/exporter/stackdriver.py [hunk 1]
```diff
@@ -12,51 +12,54 @@
...
+import time
 from opencensus.stats import aggregation as aggregation_module
-from opencensus.stats.exporters import stackdriver_exporter as stackdriver
 from opencensus.stats import measure as measure_module
 from opencensus.stats import stats as stats_module
 from opencensus.stats import view as view_module
-from opencensus.tags import tag_key as tag_key_module
+from opencensus.stats.exporters import stackdriver_exporter as stackdriver_stats
 from opencensus.tags import tag_map as tag_map_module
-from opencensus.tags import tag_value as tag_value_module
```
Comment (odeke-em) [inline examples/stats/exporter/stackdriver.py]: I think it is quite important to ensure that the exporter works with tags as the previous example had. Why this removal?

File: opencensus/stats/exporters/stackdriver_exporter.py [hunk 2]
```diff
@@ -401,19 +418,14 @@ def new_stats_exporter(options):
...
-        key = remove_non_alphanumeric(get_task_value())
-        label[key] = OPENCENSUS_TASK_DESCRIPTION
-        exporter.set_default_labels(label)
     return exporter
 
 
 def get_task_value():
     """ getTaskValue returns a task label value in the format of
-     "py-<pid>@<hostname>".
+     "py-<random_number>@<hostname>".
```
Comment (odeke-em) [inline opencensus/stats/exporters/stackdriver_exporter.py]: Why the removal of the PID? That was deterministic and the way other language exporters work.

**Comment by migibert** [comment]
Hi,
Is there any news on this one ? I just faced the issue fixed with this PR!

File: opencensus/stats/exporters/stackdriver_exporter.py [hunk 3]
```diff
@@ -197,13 +201,15 @@ def create_time_series_list(self, v_data, option_resource_type,
...
-                                                  metric_prefix)
-        set_monitored_resource(series, option_resource_type)
-
+        time_series_list = []
         tag_agg = v_data.tag_value_aggregation_data_map
         for tag_value, agg in tag_agg.items():
+            series = monitoring_v3.types.TimeSeries()
+            series.metric.type = namespaced_view_name(v_data.view.name,
+                                                      metric_prefix)
+            set_metric_labels(series, v_data.view, tag_value)
```
Comment (c24t) [inline opencensus/stats/exporters/stackdriver_exporter.py]: @mayurkale22 this looks like a bug to me, is the right behavior to set a label for each tag value (not just `tag_value`) for each series or only set a single label (in which case `set_metric_label` needs to change).

File: examples/stats/exporter/stackdriver.py (hunk 1 above)
Comment (mayurkale22) [reply examples/stats/exporter/stackdriver.py]: This effort is to make the [basic stats example](https://github.com/census-instrumentation/opencensus-java/blob/master/examples/src/main/java/io/opencensus/examples/stats/StackdriverQuickstart.java) consistent everywhere, but I am happy to rollback to previous version.

File: opencensus/stats/exporters/stackdriver_exporter.py (hunk 2 above)
Comment (mayurkale22) [reply opencensus/stats/exporters/stackdriver_exporter.py]: I agree, ```pid``` makes more sense here.

**Comment by c24t** [comment]
I addressed the rest of my comments on @mayurkale22's original PR. This change should be ready to merge, but it's likely that there are still other bugs in the stackdriver stats exporter. The existing tests aren't sufficient to show that the exporter is working.

Luckily we'll be forced to rewrite a lot of the exporter with the move to the metrics API.

**Comment by mayurkale22** [comment]
LGTM

**Pull request merged**